src/milia/api/dataset.cljc

```diff
@@ -31,33 +31,35 @@
     (parse-http :get url)))
 
 #?(:clj
-    (defn- send-file-or-params
-      "Send request with file or params"
-      [method url params suppress-4xx-exceptions?]
-      (let [options (if-let [xls_file  (:xls_file params)]
-                      (multipart-options xls_file "xls_file")
-                      {:form-params params})]
-        (parse-http method url :http-options options
-                    :suppress-4xx-exceptions? suppress-4xx-exceptions?))))
+   (defn- send-file-or-params
+     "Send request with file or params"
+     [method url params suppress-4xx-exceptions?]
+     (let [options (if-let [file (:xls_file params)]
+                     (multipart-options file "xls_file")
+                     (if-let [file (:xml_file params)]
+                       (multipart-options file "xml_file")
+                       {:form-params params}))]
```

Comment:
Can we destructure params in line 36 then replace the nested if block with the following:
```clojure
...
(cond
  (not-nil? xls_file) (multipart-options xls_file "xls_file")
  (not-nil? xml_file) (multipart-options xml_file "xml_file")
  :else {:form-params params})
```
`not-nil?` is a utility function in chimera

Reply:
awesome! refactoring...

Reply:
done! https://github.com/onaio/milia/pull/294/files#diff-b2eee2b6232252cc780df1ffc11a610a2112fe32811af815f2d61667096faadbR38-R41